plugins/modules/fabric_devices_workflow_manager.py

```diff
@@ -0,0 +1,4080 @@
+
+        Returns:
+            self (object): The current object with added L2 Handoff information.
+        Description:
+
+        """
+
+        create_l2_handoff = []
+        l2_handoff_index = -1
+
+        # The L2 Handoffs doesnot support updation
+        # So find which L2 handoffs need to be created and which needs to be ignored
+        for item in want_l2_handoff:
+            l2_handoff_index += 1
+            if not (have_l2_handoff and have_l2_handoff[l2_handoff_index]):
+                create_l2_handoff.append(item)
+
+        try:
+            payload = {"payload": create_l2_handoff}
+            task_name = "add_fabric_devices_layer2_handoffs"
+            task_id = self.get_taskid_post_api_call("sda", task_name, payload)
+            if not task_id:
+                self.msg = (
+                    "Unable to retrive the task_id for the task '{task_name}'."
+                    .format(task_name=task_name)
+                )
+                self.set_operation_result("failed", False, self.msg, "ERROR")
+                return self
+
+            success_msg = (
+                "L2 Handoff(s) '{l2_handoff}' added successfully in the device with IP '{ip}."
+                .format(l2_handoff=create_l2_handoff, ip=device_ip)
+            )
+            self.get_task_status_from_tasks_by_id(task_id, task_name, success_msg).check_return_status()
+        except Exception as msg:
+            self.msg = (
+                "Exception occured while creating the L2 Handoff(s) in the device '{ip}': {msg}"
+                .format(ip=device_ip, msg=msg)
+            )
+            self.status = "failed"
+            return self
+
+        result_fabric_device_response.update({
+            "l2_handoff_details": create_l2_handoff
+        })
+        result_fabric_device_msg.update({
+            "l2_handoff_details": "L2 Handoffs added successfully."
+        })
+
+        self.msg = "Successfully added the L2 Handoffs."
+        self.status = "success"
+        return self
+
+    def update_sda_l3_handoff(self, have_sda_l3_handoff, want_sda_l3_handoff,
+                              device_ip, result_fabric_device_response,
+                              result_fabric_device_msg):
+        """
+        Create SDA L3 Handoff in a fabric device with fields provided in playbook.
+
+        Parameters:
+            have_sda_l3_handoff (list): SDA L3 Handoff details from the Cisco Catalyst Center.
+            want_sda_l3_handoff (dict): SDA L3 Handoff details from the user playbook.
+            device_ip (str): The IP of the fabric device.
+            result_fabric_device_response (dict): The response status of the fabric device.
+            result_fabric_device_msg (dict): The message status of the fabric device.
+
+        Returns:
+            self (object): The current object with added SDA L3 Handoff information.
+        Description:
+
+        """
+
```

Comment:
self.log("Starting update process for SDA L3 Handoff on device IP: {ip}".format(ip=device_ip), "DEBUG")